formats/calendar/resources/ext.srf.formats.eventcalendar.js

```diff
@@ -201,7 +201,11 @@
 											dates.push( value.getMwTimestamp() );
 											rowData.allDay = true;
 										} else {
-											rowData.end = value.getDate().toISOString();
+											var dataDate = value.getDate();
+											if(data.query.ask.parameters.includeend && ( value.precision & 8 ) == 0 ) {
+												dataDate.setDate(dataDate.getDate() + 1);
```

Comment:
`( value.precision & 8 ) == 0`? huh? bit magic that does something... I'm sure this can be expressed more clearly

Reply:
`precision` is a bitmask set in [ext.smw.dataItem.time.js](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/res/smw/data/ext.smw.dataItem.time.js#L35), so I'll add a comment here giving a brief explanation of the bitmask.